storage-operator/pkg/salt/client.go

```diff
@@ -53,27 +53,36 @@ func NewClient(creds *Credential) *Client {
 // Spawn a job, asynchronously, to prepare the volume on the specified node.
 //
 // Arguments
-//     ctx:      the request context (used for cancellation)
-//     nodeName: name of the node where the volume will be
+//     ctx:        the request context (used for cancellation)
+//     nodeName:   name of the node where the volume will be
+//     volumeName: name of the volume to prepare
 //
 // Returns
 //     The Salt job ID.
 func (self *Client) PrepareVolume(
-	ctx context.Context, nodeName string,
+	ctx context.Context, nodeName string, volumeName string,
 ) (string, error) {
 	// Use rand_sleep to emulate slow operation for now
 	payload := map[string]interface{}{
 		"client": "local_async",
 		"tgt":    nodeName,
-		"fun":    "test.rand_sleep",
+		"fun":    "state.sls",
+		"kwarg": map[string]interface{}{
+			"mods":    "metalk8s.volumes",
+			"saltenv": "metalk8s-2.4.0-dev",
```

Comment:
Don't want to hardcode that I guess.

Should I retrieve it from the cluster (with the namespace annotations)?

Reply:
Indeed, but maybe use the target Node version and not the cluster one? Not sure though :/

Reply:
IMO the CRD (and CR, service account, ...) deployed in apiserver should be the one of the cluster_version so I think we should deploy volume using this same version

Reply:
Seen with Teddy: cluster version seems the most pertinent here.

I'll add some code to deal with that.

Reply:
Oh sorry misunderstanding what is done in the called state.
Maybe like @gdemonet said we should use the version of the node as it's mainly change on the node.

Reply:
As seen with Nicolas, we want to use the Node version, because that's the one used by the Salt minion so let's stay in sync with it.

Reply:
This is still outstanding?

Reply:
Yup, but that'll be addressed by my next commit (wasn't ready yet when I made the last push)

Reply:
Done